src/e2e-test/java/io.cdap.plugin/actions/ReplicationActions.java

```diff
@@ -0,0 +1,158 @@
+    private static String parentWindow = StringUtils.EMPTY;
+    private static final String projectId = PluginPropertyUtils.pluginProp("projectId");
+    private static final String database = PluginPropertyUtils.pluginProp("dataset");
+    public static String tableName = PluginPropertyUtils.pluginProp("sourceTable");
+    public static String schemaName = PluginPropertyUtils.pluginProp("schema");
+    public static String datatypeValues = PluginPropertyUtils.pluginProp("datatypeValuesForInsertOperation");
+    public static String deleteCondition = PluginPropertyUtils.pluginProp("deleteRowCondition");
+    public static String updateCondition = PluginPropertyUtils.pluginProp("updateRowCondition");
+    public static String updatedValue = PluginPropertyUtils.pluginProp("updatedRow");
+
+    static {
+        SeleniumHelper.getPropertiesLocators(ReplicationLocators.class);
+    }
+    public static void clickNextButton() throws InterruptedException {
+        TimeUnit.SECONDS.sleep(1);
+        ElementHelper.clickOnElement(ReplicationLocators.next);
+    }
+
+    public static void clickOnOraclePlugin() {
+        ElementHelper.clickOnElement(ReplicationLocators.oraclePlugin);
+    }
+
+    public static void selectTable() {
+        String table = schemaName + "." + tableName;
+        WaitHelper.waitForElementToBeDisplayed(ReplicationLocators.selectTable(table), 300);
+        AssertionHelper.verifyElementDisplayed(ReplicationLocators.selectTable(table));
+        ElementHelper.clickOnElement(ReplicationLocators.selectTable(table));
+    }
+
+    public static void deployPipeline() {
+        ElementHelper.clickOnElement(ReplicationLocators.deployPipeline);
+    }
+
+    public static void startPipeline() {
+        ElementHelper.clickIfDisplayed(ReplicationLocators.start, ConstantsUtil.DEFAULT_TIMEOUT_SECONDS);
+    }
+
+    public static void runThePipeline() {
+        startPipeline();
+        WaitHelper.waitForElementToBeDisplayed(ReplicationLocators.running);
+    }
+
+    public static void openAdvanceLogs() {
+        ReplicationLocators.logs.click();
+        parentWindow = SeleniumDriver.getDriver().getWindowHandle();
+        ArrayList<String> tabs = new ArrayList(SeleniumDriver.getDriver().getWindowHandles());
+        SeleniumDriver.getDriver().switchTo().window(tabs.get(tabs.indexOf(parentWindow) + 1));
+        ReplicationLocators.advancedLogs.click();
+    }
+
+    public static void captureRawLog() {
+        //Capturing raw logs.
+        try {
+            String rawLogs = getRawLogs();
+            String logsSeparatorMessage = ConstantsUtil.LOGS_SEPARATOR_MESSAGE
+                    .replace("MESSAGE", "DEPLOYED PIPELINE RUNTIME LOGS");
+            BeforeActions.scenario.write(rawLogs);
+            CdfPipelineRunAction.writeRawLogsToFile(BeforeActions.file, logsSeparatorMessage, rawLogs);
+        } catch (Exception e) {
+            BeforeActions.scenario.write("Exception in capturing logs : " + e);
```

Comment:
Do we want to fail the test here?

Reply:
I don't think we should fail the test on this, because replication happened successfully, else it would fail before.This should suffice.